### PR TITLE
Lunchtime12 dash

### PR DESCRIPTION
**Pull request by GwydionJon**
Introduction to dash for interactive data visualization and UI building.

### CODE REVIEW

File: lunchtime12/lunchtime12.ipynb [hunk 1]
```diff
@@ -0,0 +1,757 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Welcome to Dash\n",
+    "\n",
+    "UI elements not quite as messy as ipywidgets.\n",
```
Comment (iulusoy) [inline lunchtime12/lunchtime12.ipynb:14]: Maybe a bit more positive or state the actual difference

File: lunchtime12/lunchtime12.ipynb [hunk 2]
```diff
@@ -0,0 +1,757 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "notes"
+    }
+   },
+   "source": [
+    "Simple text is automatically shown side by side: <br>\n",
```
Comment (iulusoy) [inline lunchtime12/lunchtime12.ipynb:31]: this comes a bit abruptly, maybe something general about the workings of Dash, that it is on top of Flask etc and that here you use the div component to interact

File: lunchtime12/lunchtime12.ipynb [hunk 3]
```diff
@@ -0,0 +1,757 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import plotly.express as px\n",
```
Comment (iulusoy) [inline lunchtime12/lunchtime12.ipynb:49]: You skip this slide but `np` is required later

File: lunchtime12/lunchtime12.ipynb [hunk 4]
```diff
@@ -0,0 +1,757 @@
...
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Using dash inside a class\n",
+    "\n",
+    "Unfortunately the decorator style of dashs callbacks we have used so far is very much incompatible with encapsulating the dash app inside a class. \n",
+    "Normally the app itself is to be used in the whole module. <br>\n",
+    "`@self.app.callback()` or smililar things don't work.\n",
+    "However we can simply refer any function as a calback as seen here:\n",
```
Comment (iulusoy) [inline lunchtime12/lunchtime12.ipynb:449]: typo `calback` should be `callback`

**Comment by iulusoy** [comment]
@lkeegan `numba` seems to require `scipy`, see [here](https://github.com/numba/numba/blob/bca99bcde4abf9df48af9095ce4d0e4d6d76df80/numba/__init__.py#L13), but previously the tests ran without it (and it was the same `numba` version, `0.57.0`. Installing `scipy` into the environment makes the notebook run though.

**Comment by lkeegan** [comment]
@iulusoy it looks like for numba scipy is an optional dependency, but if present it must be version >=1.0.0
Maybe you had an older version of scipy installed? Or are there failing tests in CI or somewhere else that I'm not seeing?

**Comment by iulusoy** [comment]
> @iulusoy it looks like for numba scipy is an optional dependency, but if present it must be version >=1.0.0 Maybe you had an older version of scipy installed? Or are there failing tests in CI or somewhere else that I'm not seeing?

Even with no scipy in the environment the notebook does not run with `AttributeError: module 'scipy' has no attribute '__version__'` since the scipy version is checked but no scipy present.

CI was running but if I run the tests locally in a new environment this error pops up. Have not tried the CI on GitHub yet but will trigger it now.

File: requirements.in [hunk 5]
```diff
@@ -12,6 +12,7 @@ ipytest
 pillow
 jinja2
 numba
+scipy
```
Comment (iulusoy) [inline requirements.in:15]: so we do need scipy now for the CI to run? I think the run without it was cancelled.

**Review by iulusoy — APPROVED** [review]
lgtm

**Pull request merged**